packages/driver/cypress/integration/commands/actions/type_spec.js

```diff
@@ -604,6 +604,94 @@ describe('src/cy/commands/actions/type - #type', () => {
     })
   })
 
+  describe(`type(' ') fires click event on button-like elements`, () => {
+    beforeEach(() => {
+      cy.visit('fixtures/click-event-by-type.html')
+    })
+
+    const targets = [
+      'button-tag',
+      'input-button',
+      'input-image',
+      'input-reset',
+      'input-submit',
+    ]
+
+    describe(`triggers with single ' '`, () => {
+      targets.forEach((targetId) => {
+        it(`${targetId}`, () => {
+          cy.get(`#target-${targetId}`).focus()
+          cy.get(`#target-${targetId}`).type(' ')
+
+          cy.get('li').eq(0).should('have.text', 'keydown')
+          cy.get('li').eq(1).should('have.text', 'keypress')
+          cy.get('li').eq(2).should('have.text', 'keyup')
+          cy.get('li').eq(3).should('have.text', 'click')
+        })
+      })
+    })
+
+    describe('triggers after other characters', () => {
+      targets.forEach((targetId) => {
+        it(`${targetId}`, () => {
+          cy.get(`#target-${targetId}`).focus()
+          cy.get(`#target-${targetId}`).type('asd ')
+
+          cy.get('li').eq(12).should('have.text', 'click')
+        })
+      })
+    })
+
+    describe('checkbox', () => {
+      it('checkbox is checked/unchecked', () => {
+        cy.get(`#target-input-checkbox`).focus()
+        cy.get(`#target-input-checkbox`).type(' ')
+
+        cy.get('li').eq(0).should('have.text', 'keydown')
+        cy.get('li').eq(1).should('have.text', 'keypress')
+        cy.get('li').eq(2).should('have.text', 'keyup')
+        cy.get('li').eq(3).should('have.text', 'click')
+
+        cy.get('#target-input-checkbox').should('be.checked')
+
+        cy.get(`#target-input-checkbox`).type(' ')
+
+        cy.get('li').eq(4).should('have.text', 'keydown')
+        cy.get('li').eq(5).should('have.text', 'keypress')
+        cy.get('li').eq(6).should('have.text', 'keyup')
+        cy.get('li').eq(7).should('have.text', 'click')
+
+        cy.get('#target-input-checkbox').should('not.be.checked')
+      })
+    })
+
+    describe('radio', () => {
+      it('radio fires click event when it is not checked', () => {
+        cy.get(`#target-input-radio`).focus()
+        cy.get(`#target-input-radio`).type(' ')
+
+        cy.get('li').eq(0).should('have.text', 'keydown')
+        cy.get('li').eq(1).should('have.text', 'keypress')
+        cy.get('li').eq(2).should('have.text', 'keyup')
+        cy.get('li').eq(3).should('have.text', 'click')
+
+        cy.get('#target-input-radio').should('be.checked')
+      })
+
+      it('radio does not fire click event when it is checked', () => {
```

Comment:
This test does appear to fire a click event? Also is this relying on the state of `radio fires click event when it is not checked`? If not, it appears to be the same tests as `radio fires click event when it is not checked`.

Reply:
Radio button has 2 states: unchecked (empty circle), checked (filled circle). When you focus the empty radio button (unchecked) and press the space bar, it is filled (checked) and `click` event is sent. 

But when we press the space bar again, `click` event is not fired because the radio button is already checked. 

To make it clear, I've edited the code like below:

```js
// We're clicking here first to make the radio element checked.
cy.get(`#target-input-radio`).click().type(' ')

// item 0 is click event. It's fired because we want to make sure our radio button is checked.
cy.get('li').eq(1).should('have.text', 'keydown')
cy.get('li').eq(2).should('have.text', 'keypress')
cy.get('li').eq(3).should('have.text', 'keyup')

cy.get('#target-input-radio').should('be.checked')
```